src/features/smartSearch/components/filters/EmailHistory/index.tsx

```diff
@@ -0,0 +1,208 @@
+
+    setConfig({
+      ...rest,
+      ...(range.after && { after: range.after }),
+      ...(range.before && { before: range.before }),
+    });
+  };
+
+  return (
+    <FilterForm
+      disableSubmit={
+        (emailSelectScope === EMAIL_SELECT_SCOPE.FROM_PROJECT &&
+          !filter.config.campaign) ||
+        (emailSelectScope === EMAIL_SELECT_SCOPE.SPECIFIC_EMAIL &&
+          !filter.config.email)
+      }
+      onCancel={onCancel}
+      onSubmit={(e) => {
+        e.preventDefault();
+        onSubmit(filter);
+      }}
+      renderSentence={() => (
+        <Msg
+          id={localMessageIds.inputString}
+          values={{
+            addRemoveSelect: (
+              <StyledSelect
+                onChange={(e) => setOp(e.target.value as OPERATION)}
+                value={filter.op}
+              >
+                {Object.values(OPERATION).map((o) => (
+                  <MenuItem key={o} value={o}>
+                    <Msg id={messageIds.operators[o]} />
+                  </MenuItem>
+                ))}
+              </StyledSelect>
+            ),
+            emailScopeSelect: (
+              <StyledSelect
+                onChange={(e) => {
+                  removeKey(['email', 'campaign']);
+                  setEmailSelectScope(e.target.value as EMAIL_SELECT_SCOPE);
+                }}
+                value={emailSelectScope}
+              >
+                {Object.values(EMAIL_SELECT_SCOPE).map((item) => (
+                  <MenuItem key={item} value={item}>
+                    <Msg id={messageIds.filters.emailScopeSelect[item]} />
+                  </MenuItem>
+                ))}
+              </StyledSelect>
+            ),
+            emailSelect:
+              emailSelectScope === EMAIL_SELECT_SCOPE.SPECIFIC_EMAIL ? (
+                <StyledSelect
+                  onChange={(e) =>
+                    setValueToKey('email', parseInt(e.target.value))
+                  }
+                  value={filter.config.email || ''}
+                >
+                  {emailsFuture?.map((email) => (
+                    <MenuItem key={`email-${email.id}`} value={email.id}>
+                      {`"${email.title}"`}
+                    </MenuItem>
+                  ))}
+                </StyledSelect>
+              ) : null,
+            operatorSelect: (
+              <StyledSelect
+                onChange={(e) => setValueToKey('operator', e.target.value)}
+                value={filter.config.operator}
+              >
+                {Object.values(EMAIL_HISTORY_OP).map((status) => (
+                  <MenuItem key={status} value={status}>
+                    <Msg
+                      id={
+                        localMessageIds.operatorSelect[
+                          convertToMessageKey(status) as
+                            | 'notSent'
+                            | 'opened'
+                            | 'sent'
+                            | 'notOpened'
```

Comment:
Here's an alternative solution you could use (but I would use the same solution both places), so pick just one:

```tsx
// Inside the component
const messages = useMessage(localMessageIds);
const MESSAGE_BY_OP = {
  not_opened: messages.operatorSelect.notOpened(),
  not_sent: messages.operatorSelect.notSent(),
  opened: messages.operatorSelect.opened(),
  sent: messages.operatorSelect.sent(),
} as const;
```